app/services/activitypub/fetch_replies_service.rb

```diff
@@ -54,10 +69,14 @@ def fetch_collection(collection_or_uri)
   end
 
   def filtered_replies
-    # Only fetch replies to the same server as the original status to avoid
-    # amplification attacks.
+    if @filter_by_host
+      # Only fetch replies to the same server as the original status to avoid
+      # amplification attacks.
 
-    # Also limit to 5 fetched replies to limit potential for DoS.
-    @items.map { |item| value_or_id(item) }.reject { |uri| non_matching_uri_hosts?(@account.uri, uri) }.take(5)
+      # Also limit to 5 fetched replies to limit potential for DoS.
+      @items.map { |item| value_or_id(item) }.reject { |uri| non_matching_uri_hosts?(@account.uri, uri) }.take(MAX_REPLIES)
+    else
+      @items.map { |item| value_or_id(item) }.take(MAX_REPLIES)
```

Comment:
while this is unnecessary since the subclass overrides `filtered_replies`, it is added for consistency, since the instance variable "filter_by_host" used to switch this behavior in `fetch_collection` implies that it would do the same thing here.